src/TestableIO.System.IO.Abstractions.TestingHelpers/MockDirectory.cs

```diff
@@ -503,7 +503,8 @@ public override void Move(string sourceDirName, string destDirName)
         var fullSourcePath = mockFileDataAccessor.Path.GetFullPath(sourceDirName).TrimSlashes();
         var fullDestPath = mockFileDataAccessor.Path.GetFullPath(destDirName).TrimSlashes();
 
-        if (mockFileDataAccessor.StringOperations.Equals(fullSourcePath, fullDestPath))
+        if (XFS.IsUnixPlatform() ? mockFileDataAccessor.StringOperations.Equals(fullSourcePath, fullDestPath)
```

Comment:
When `XFS.IsUnixPlatform()` the [`StringOperations` ](https://github.com/TestableIO/System.IO.Abstractions/blob/main/src/TestableIO.System.IO.Abstractions.TestingHelpers/StringOperations.cs) already use `StringComparison.Ordinal`, so this check could be simplified to
```csharp
if (string.Equals(fullSourcePath, fullDestPath, StringComparison.Ordinal))
```